### PR TITLE
Feature/functional locations

### PR DESCRIPTION
**Pull request by gecko17**
# Description

Introduced functionality for retrieving functional locations from asset central. The functionality of functional locations was introduced into AC and PAI in later releases. Functional Locations can be used in almost the same way as Equipment in PAI. For some customer scenarios it is required to work with Functional Locations in the same way as with Equipment. Hence sailor should provide fetching the relevant information.

This feature covers the basic functionality for retrieving functional locations with the filter language used in the other methods. More advanced functionality, similar to the equipment model could be implemented further.

# Additional information

Functionality was tested locally. Unit Tests not included. Documentation was updated. All implemented methods include docstrings.

# Checklist

Mark "not applicable" if item on the list does not apply to this pull request.

- [x] I have created/adapted unit tests for new code
  - [ ] not applicable 
- [ ] I have added new dependencies as described at the [Contributing](https://sap.github.io/project-sailor/contributing.html#requirements-management) page
  - [x] not applicable 
- [x] I have made corresponding changes to the documentation (incl. tutorial, etc.)
  - [ ] not applicable 
- [x] I have provided release notes (in description or as comment) if this PR is a new feature OR a change worth mentioning for next release
  - [ ] not applicable 
- [x] I have obtained API approvals if a new SAP API or endpoint is used
  - [ ] not applicable

### CODE REVIEW

File: sailor/assetcentral/functional_location.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+from sailor import _base
+from ..utils.timestamps import _string_to_timestamp_parser
+from .utils import _ac_fetch_data, AssetcentralEntity, _AssetcentralField, AssetcentralEntitySet, \
+    _ac_application_url
+from .constants import VIEW_FUNCTIONAL_LOCATIONS
+
+
+_FUNCTIONAL_LOCATION_FIELDS = [
+    _AssetcentralField('name', 'name'),
+    _AssetcentralField('internal_id', 'internalId'),
```
Comment (atrejut) [inline sailor/assetcentral/functional_location.py:17]: at least for `Equipment` there's both `name` and `internalId` but they always return the exact same value, so we're returning only one and ignoring the other. Is it the same here, maybe?

File: sailor/assetcentral/functional_location.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        find_functional_locations(name=['MyFloc', 'MyOtherFloc'])
+
+    Find all Functional Locations with the name 'MyFloc' which are also located in 'London'::
+
+        find_functional_locations(name='MyFloc', location_name='London')
+
+    Find all Functional Locations created between January 1, 2018 and January 1, 2019 in 'London'::
+
+        find_functional_locations(extended_filters=['_created_on >= "2018-01-01"', '_created_on < "2019-01-01"'],
```
Comment (atrejut) [inline sailor/assetcentral/functional_location.py:115]: I'd prefer not to have any 'private' fields (fields starting with an `_` are not part of the stable API) in these documentation examples.
Two options:
- just remove this last example, I don't think it's necessary
- If you have a use-case that actually requires `_created_on` rename the field in the field map to `created_on`

**Review by atrejut — COMMENTED** [review]
looks very good 👍 
just two minor questions from my side

File: sailor/assetcentral/functional_location.py (hunk 1 above)
Comment (gecko17) [reply sailor/assetcentral/functional_location.py:17]: I guess it is the same behavior here. At least I can confirm for some samples I checked.

**Review by atrejut — APPROVED** [review]
I'm good with this I think. @cloudyday

**Review by cloudyday — APPROVED** [review]
LGTM. Will work in my comments in separate PR.

**Pull request merged**